### ISSUE TITLE
Test Results

### ISSUE THREAD

**Issue by VisionLover** [haoningwu3639/StoryGen#34]
Hi,

First of all, thank you for your excellent work on this project. I have a few questions that I hope you can help clarify:

1. **Test Results:** I tested your code on the provided test dataset and observed that the CLIP-I and CLIP-T results showed only a 1% variance, which seems consistent. However, I noticed that the FID score was significantly different from what I expected. Using the same `pytorch-fid` tool, I obtained an FID score of 86.27. Could you provide any insights into why this discrepancy might be occurring?

2. **Finetuned SDM Results:** I noticed there is a result published in your paper titled `finetuned-sdm`. Based on my understanding of the paper, it seems that your checkpoint should be used with `stage=no` set. Is this interpretation correct?

3. **Variance of Results:** Lastly, could you provide some information on the general variance of the results? How consistent are the outcomes typically?

Thank you in advance for your time and response!

**Comment by haoningwu3639** [comment]
1. I think there might be several potential possibilities: (i) The most likely reason is that in the *inference* code we provided, we did not include the code for using **PickScore** to filter the generated candidate images within the Sampler, as described in our paper. This was done for general usability purposes. You can refer to the code provided in the evaluation folder and add this to the Sampler to filter the best results. (ii) To accommodate the settings of AR-LDM and Story DALL·E for Story Continuation, we used the previous frame from the paired data in the StorySalon test set as the reference image input. Please ensure the correctness of the dataloader.
2. Your understanding is not accurate. As described in our paper, setting the stage to “no” corresponds to our StoryGen-Single variant, while the finetuned-sdm refers to the model where all SDM parameters have been trained on our StorySalon dataset.
3. As discussed in point 1, we applied the same test settings to all the models, specifically generating 10 samples and using PickScore to select the best candidate. Therefore, we did not consider the variance in metrics due to multiple experiments in our study (as our evaluation results show, the variance is minimal).

**Comment by VisionLover** [comment]
Thank you for your reply.

(i) While it's true that `PickScore` isn't utilized in the `inference.py` code, I noticed that it is indeed used in the `inference_COCO.py` script. In that script, only the best image out of the 10 generated images is saved. I've also adapted this code slightly to work with the StorySalon dataset.

(ii) Additionally, I modified the number of previous images to one in the `dataset.py` file. I then created the data loader from the test set using this adjusted code.

**Comment by VisionLover** [comment]
Thank you for your reply.

(i) While it's true that PickScore isn't utilized in the inference.py code, I noticed that it is indeed used in the inference_COCO.py script. In that script, only the best image out of the 10 generated images is saved. I've also adapted this code slightly to work with the StorySalon dataset.

(ii) Additionally, I modified the number of previous images to one in the dataset.py file. I then created the data loader from the test set using this adjusted code.

@haoningwu3639

**Comment by haoningwu3639** [comment]
You may also need to be aware that when calculating the FID, we only use the corresponding ground truth frames in the test set as references. 
Additionally, since the number of test samples may influence the FID metric, its absolute value can sometimes be less significant. Its relative value, however, can reflect the merits and demerits of the model. 
Considering the metrics from CLIP, the model and checkpoint you loaded appear to be correct. I believe that the value of the FID may have been affected by the number of test samples.
